### PR TITLE
Allowing failure with different types of error

### PR DESCRIPTION
**Pull request by oxtoacart**
For getlantern/lantern-internal#3537

### CODE REVIEW

File: ops_test.go [hunk 1]
```diff
@@ -57,6 +58,7 @@ func doTestFailure(t *testing.T, cancel bool) {
 	var wg sync.WaitGroup
 	wg.Add(1)
 	op.Go(func() {
+		op.FailIf(gerrors.New("This error should be ignored"))
```
Comment (hwh33) [inline ops_test.go:61]: Why does this one get ignored? Just curious.

**Comment by oxtoacart** [comment]
Thanks @hwh33 !

File: ops_test.go (hunk 1 above)
Comment (oxtoacart) [reply ops_test.go:61]: Because FailIf only records the last error given to it.

Comment (hwh33) [reply ops_test.go:61]: Ahhh. D'oh 🤦‍♂️

Thanks!

**Pull request merged**